libcextract/ElfCXX.cpp

```diff
@@ -115,6 +133,92 @@ ElfObject::~ElfObject(void)
   }
 }
 
+Elf *ElfObject::decompress_gz(std::string &parser_path)
+{
+  const size_t CHUNK = 16384;
+
+  unsigned have;
+  unsigned char in[CHUNK];
+  unsigned char out[CHUNK];
+
+  unsigned long dest_size = CHUNK;
+  unsigned char *dest = (unsigned char *)malloc(CHUNK);
+  if (!dest)
+    throw std::runtime_error("zlib dest malloc failed\n");
+
+  unsigned long dest_current = 0;
+
+  std::cout << "Extrating gzipped file: " + parser_path + "\n";
+
+  int fd = open(parser_path.c_str(), O_RDONLY);
```

Comment:
This is the ElfFd. There is no need to open this file again. It is already opened.